gemrb/core/GameScript/Triggers.cpp

```diff
@@ -2088,7 +2088,16 @@ int GameScript::CheckStat(Scriptable* Sender, Trigger* parameters)
 		return 0;
 	}
 	Actor* actor = ( Actor* ) target;
-	if ( (signed) actor->GetStat( parameters->int1Parameter ) == parameters->int0Parameter) {
+	int stat;
+	// hack for mods that reuse 'empty' stats 176-182, corresponding to scripting states 20-26
+	if (core->HasFeature(GF_FAKE_SCRIPTING_STATES)
+	    && (parameters->int1Parameter >= IE_SCRIPTINGSTATE1 + 20)
+	    && (parameters->int1Parameter <= IE_SCRIPTINGSTATE1 + 26)) {
+		stat = actor->fakeScriptingStates[parameters->int1Parameter - IE_SCRIPTINGSTATE1 - 20];
```

Comment:
Indeed a hack! I forgot why Avenger wanted to keep the stat count below 255, but there are pst-only stats that can be used for this instead. The first two of the current list and surely more, but there's a simpler candidate: just refactor  IE_SPLSTATE_ID1 as the TODO notes. A simple thing, since it's not a saved stat and only has a few users.

Reply:
FWIW, there are only six IWD spell states under 255, and we need seven.  But other than that, I agree that moving spell states would look neater.  We can even dynamically allocate them based on  `SPLSTATE.IDS` length, like how ToBEx does with its 401+ stats.

Reply:
You can reuse also the existing 176/177, since they're pst-only.